curvefs/src/mds/fs_manager.cpp

```diff
@@ -253,11 +254,18 @@ FSStatusCode FsManager::CreateFs(const ::curvefs::mds::CreateFsRequest* request,
     const auto& fsType = request->fstype();
     const auto& detail = request->fsdetail();
 
+    // check fsname
+    if (!std::regex_match(fsName.cbegin(), fsName.cend(),
+                          std::regex(FSNAME_PATTERN))) {
```

Comment:
Does creating a regex object have runtime cost? If so, it's better to make it a static object, because the pattern is fixed.